cmd/kubeadm/app/util/version.go

```diff
@@ -144,6 +144,14 @@ func KubernetesIsCIVersion(version string) bool {
 	if len(subs) == 1 && len(subs[0]) == 4 && strings.HasPrefix(subs[0][2], "ci") {
 		return true
 	}
+	ver, err := versionutil.ParseSemantic(version)
+	if err != nil {
+		klog.Warningf("could not parse semantic version %s: %v", version, err)
+		return false
+	}
+	if len(ver.PreRelease()) != 0 {
```

Comment:
```suggestion
	if len(ver.PreRelease()) != 0 && ver != constants.DefaultKubernetesPlaceholderVersion {
```
`DefaultKubernetesPlaceholderVersion` is now `defaultKubernetesPlaceholderVersion`

v1.0.0-placeholder-version is a place holder, not ci version.

Reply:
```
diff --git a/cmd/kubeadm/app/constants/constants.go b/cmd/kubeadm/app/constants/constants.go
index 124295d0512..87342950fc1 100644
--- a/cmd/kubeadm/app/constants/constants.go
+++ b/cmd/kubeadm/app/constants/constants.go
@@ -508,9 +508,9 @@ var (
                Jitter:   0.1,
        }
 
-       // defaultKubernetesPlaceholderVersion is a placeholder version in case the component-base
+       // DefaultKubernetesPlaceholderVersion is a placeholder version in case the component-base
        // version was not populated during build.
-       defaultKubernetesPlaceholderVersion = version.MustParseSemantic("v1.0.0-placeholder-version")
+       DefaultKubernetesPlaceholderVersion = version.MustParseSemantic("v1.0.0-placeholder-version")
 )
 
 // getSkewedKubernetesVersion returns the current MAJOR.(MINOR+n).0 Kubernetes version with a skew of 'n'
@@ -528,7 +528,7 @@ func getSkewedKubernetesVersionImpl(versionInfo *apimachineryversion.Info, n int
        // More changes would be required if the kubelet version one day decouples from that of Kubernetes.
        var ver *version.Version
        if len(versionInfo.Major) == 0 {
-               return defaultKubernetesPlaceholderVersion
+               return DefaultKubernetesPlaceholderVersion
        }
        ver = version.MustParseSemantic(versionInfo.GitVersion)
        // Append the MINOR version skew.
diff --git a/cmd/kubeadm/app/constants/constants_test.go b/cmd/kubeadm/app/constants/constants_test.go
index 97046c592a3..afa68689911 100644
--- a/cmd/kubeadm/app/constants/constants_test.go
+++ b/cmd/kubeadm/app/constants/constants_test.go
@@ -244,7 +244,7 @@ func TestGetSkewedKubernetesVersionImpl(t *testing.T) {
                {
                        name:           "invalid versionInfo; placeholder version is returned",
                        versionInfo:    &apimachineryversion.Info{},
-                       expectedResult: defaultKubernetesPlaceholderVersion,
+                       expectedResult: DefaultKubernetesPlaceholderVersion,
                },
                {
                        name:           "valid skew of -1",
diff --git a/cmd/kubeadm/app/util/version.go b/cmd/kubeadm/app/util/version.go
index 744c4a810fa..a50d5d8f43e 100644
--- a/cmd/kubeadm/app/util/version.go
+++ b/cmd/kubeadm/app/util/version.go
@@ -149,7 +149,7 @@ func KubernetesIsCIVersion(version string) bool {
                klog.Warningf("could not parse semantic version %s: %v", version, err)
                return false
        }
-       if len(ver.PreRelease()) != 0 {
+       if len(ver.PreRelease()) != 0 && ver.String() != constants.DefaultKubernetesPlaceholderVersion.String() {
                return true
        }
        return false
```